src/components/Notifications/JoinColumnSelectionTable/JoinColumnSelectionTable.js

```diff
@@ -0,0 +1,214 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import './JoinColumnSelectionTable.scss'
+import { Select } from '../../Select'
+import { Icon } from '../../Icon'
+import { isListQuery, isAggregation, getGroupableColumns, getStringColumnIndices } from 'autoql-fe-utils'
+export default class JoinColumnSelectionTable extends React.Component {
+  constructor(props) {
+    super(props)
+    const selectableJoinColumnsLength = this.getSelectableJoinColumnsLength()
+    this.state = {
+      showSecondRow: false,
+      selectableJoinColumnsLength: selectableJoinColumnsLength,
+      firstQueryFirstValue: props.storedInitialData?.[0]?.join_columns?.[0] || props.firstQueryFirstValue,
+      firstQuerySecondValue: props.storedInitialData?.[0]?.join_columns?.[1] || props.firstQuerySecondValue,
+      secondQueryFirstValue: props.storedInitialData?.[1]?.join_columns?.[0] || props.secondQueryFirstValue,
+      secondQuerySecondValue: props.storedInitialData?.[1]?.join_columns?.[1] || props.secondQuerySecondValue,
+    }
+  }
+
+  static propTypes = {
+    storedInitialData: PropTypes.arrayOf(
+      PropTypes.shape({
+        join_columns: PropTypes.arrayOf(PropTypes.string),
+      }),
+    ),
+    columnHeaders: PropTypes.arrayOf(PropTypes.string),
+    rowHeaders: PropTypes.arrayOf(PropTypes.string),
+    firstQueryResult: PropTypes.shape({}),
+    secondQueryResult: PropTypes.shape({}),
+    firstQueryFirstValue: PropTypes.string,
+    firstQuerySecondValue: PropTypes.string,
+    secondQueryFirstValue: PropTypes.string,
+    secondQuerySecondValue: PropTypes.string,
+    onFirstQueryFirstValueChange: PropTypes.func,
+    onFirstQuerySecondValueChange: PropTypes.func,
+    onSecondQueryFirstValueChange: PropTypes.func,
+    onSecondQuerySecondValueChange: PropTypes.func,
```

Comment:
Does you need prop types for these values (line 31-38) if they are just state values?

Reply:
I do need those prop types as I do have 
this.props.firstQueryFirstValue and so on. At least it won't hurt anything?
<JoinColumnSelectionTable
          columnHeaders={['Query 1', 'Query 2']}
          rowHeaders={['1st Join', '2nd Join']}
          firstQueryFirstOptions={firstQuerySelectableJoinColumns.map((col) => ({
            value: col.name,
            label: col.display_name,
          }))}
          firstQuerySecondOptions={firstQuerySelectableJoinColumns.map((col) => ({
            value: col.name,
            label: col.display_name,
          }))}
          secondQueryFirstOptions={secondQuerySelectableJoinColumns.map((col) => ({
            value: col.name,
            label: col.display_name,
          }))}
          secondQuerySecondOptions={secondQuerySelectableJoinColumns.map((col) => ({
            value: col.name,
            label: col.display_name,
          }))}
          firstQueryResult={this.state.firstQueryResult}
          secondQueryResult={this.state.secondQueryResult}
          firstQueryFirstValue={this.state.firstQueryFirstValue}
          firstQuerySecondValue={this.state.firstQuerySecondValue}
          secondQueryFirstValue={this.state.secondQueryFirstValue}
          secondQuerySecondValue={this.state.secondQuerySecondValue}
          onFirstQueryFirstValueChange={this.handleFirstQueryFirstValueChange}
          onFirstQuerySecondValueChange={this.handleFirstQuerySecondValueChange}
          onSecondQueryFirstValueChange={this.handleSecondQueryFirstValueChange}
          onSecondQuerySecondValueChange={this.handleSecondQuerySecondValueChange}
          onRemoveSecondValues={this.handleRemoveSecondValues}
          isLoadingSecondQuery={this.state.isLoadingSecondQuery}
          storedInitialData={this.state.storedInitialData}
        />